src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,39 @@
-package core.basesyntax;
-
-public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
-    }
-}
+package core.basesyntax;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+public class SalaryInfo {
+    private static final SimpleDateFormat SDF = new SimpleDateFormat("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
+        StringBuilder getSalaryInfo = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo);
+        int[] salariesInfo = new int[names.length];
+
+        for (int i = 0; i < names.length; i++) {
+            for (String datum : data) {
+                if (dateNotInRange(datum.split(" ")[0], dateFrom, dateTo)) {
+                    continue;
+                }
+                if (datum.split(" ")[1].equals(names[i])) {
+                    salariesInfo[i] += Integer.parseInt(datum.split(" ")[2])
+                            * Integer.parseInt(datum.split(" ")[3]);

```

Comment:
Any magic numbers should be constants (https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)
Your code should be easy to read. Move all hardcoded values to constant fields and give them informative names.
Separator " - " can be constant too.